src/transformers/models/seggpt/image_processing_seggpt.py

```diff
@@ -0,0 +1,440 @@
+
+        data = {}
+
+        if images is not None:
+            images = self._preprocess_step(
+                images,
+                do_resize=do_resize,
+                size=size,
+                resample=resample,
+                do_rescale=do_rescale,
+                rescale_factor=rescale_factor,
+                do_normalize=do_normalize,
+                image_mean=image_mean,
+                image_std=image_std,
+                data_format=data_format,
+                input_data_format=input_data_format,
+                **kwargs,
+            )
+
+            data["pixel_values"] = images
+
+        if prompt_images is not None:
+            prompt_images = self._preprocess_step(
+                prompt_images,
+                do_resize=do_resize,
+                size=size,
+                resample=resample,
+                do_rescale=do_rescale,
+                rescale_factor=rescale_factor,
+                do_normalize=do_normalize,
+                image_mean=image_mean,
+                image_std=image_std,
+                data_format=data_format,
+                input_data_format=input_data_format,
+                **kwargs,
+            )
+
+            data["prompt_pixel_values"] = prompt_images
+
+        if prompt_masks is not None:
+            prompt_masks = self._preprocess_step(
+                prompt_masks,
+                do_resize=do_resize,
+                size=size,
+                resample=PILImageResampling.NEAREST,
+                do_rescale=do_rescale,
+                rescale_factor=rescale_factor,
+                do_normalize=do_normalize,
+                image_mean=image_mean,
+                image_std=image_std,
+                data_format=data_format,
+                input_data_format=input_data_format,
+                **kwargs,
+            )
+
+            data["prompt_masks"] = prompt_masks
+
+        return BatchFeature(data=data, tensor_type=return_tensors)
+
+    def post_process_masks(
+        self, outputs, target_sizes: Optional[List[Tuple[int, int]]] = None
+    ) -> List[Dict[str, TensorType]]:
+        """
+        Converts the output of [`SegGptImageSegmentationOutput`] into segmentation maps. Only supports
+        PyTorch.
+
+        Args:
+            outputs ([`SegGptImageSegmentationOutput`]):
+                Raw outputs of the model.
+            target_sizes (`List[Tuple[int, int]]`, *optional*):
+                List of length (batch_size), where each list item (`Tuple[int, int]`) corresponds to the requested
+                final size (height, width) of each prediction. If left to None, predictions will not be resized.
+        Returns:
+            `List[Dict[str, TensorType]]`: A list of dictionaries, each dictionary containing the mask for an image
+            in the batch as predicted by the model.
+        """
+        masks = outputs.pred_masks  # batch_size x num_channels x 2*height x width
```

Comment:
To raise error if torch not available 

```suggestion
        requires_backends(self, ["torch"])
        masks = outputs.pred_masks  # batch_size x num_channels x 2*height x width
```